server/controllers/recommendation/bookToPpudController.ts

```diff
@@ -83,53 +81,74 @@ async function post(req: Request, res: Response, _: NextFunction) {
-          sentences: [],
-          status: 'N/A',
-          youngOffender: 'N/A',
-        },
-      },
-      token,
-      featureFlags: flags,
-    })
+      await ppudUpdateOffender(token, offenderId, {
+        nomsId: recommendation.personOnProbation.nomsNumber,
+        croNumber: recommendation.personOnProbation.croNumber,
+        dateOfBirth: recommendation.bookRecallToPpud?.dateOfBirth,
+        ethnicity: recommendation.bookRecallToPpud?.ethnicity,
+        firstNames: recommendation.bookRecallToPpud?.firstNames,
+        familyName: recommendation.bookRecallToPpud?.lastName,
+        gender: recommendation.bookRecallToPpud?.gender,
+        isInCustody,
+        prisonNumber: recommendation.bookRecallToPpud?.prisonNumber,
+        address,
+        additionalAddresses,
+      })
+    } else {
+      const createOffenderResponse = await ppudCreateOffender(token, {
+        nomsId: recommendation.personOnProbation.nomsNumber,
+        croNumber: recommendation.personOnProbation.croNumber,
+        custodyType: recommendation.bookRecallToPpud?.custodyType,
+        dateOfBirth: recommendation.bookRecallToPpud?.dateOfBirth,
+        dateOfSentence: recommendation.bookRecallToPpud?.sentenceDate,
+        ethnicity: recommendation.bookRecallToPpud?.ethnicity,
+        firstNames: recommendation.bookRecallToPpud?.firstNames,
+        familyName: recommendation.bookRecallToPpud?.lastName,
+        gender: recommendation.bookRecallToPpud?.gender,
+        isInCustody,
+        indexOffence: recommendation.bookRecallToPpud?.indexOffence,
+        mappaLevel: recommendation.bookRecallToPpud?.mappaLevel,
+        prisonNumber: recommendation.bookRecallToPpud?.prisonNumber,
+        address,
+        additionalAddresses,
+      })
 
-    const offenderId = createOffenderResponse.offender.id
-    const sentenceId = createOffenderResponse.offender.sentence.id
+      offenderId = createOffenderResponse.offender.id
+      sentenceId = createOffenderResponse.offender.sentence.id
+
+      // write ppudOffender details, so that create offender is never called again, but rather update offender.
+      await updateRecommendation({
+        recommendationId: String(recommendationId),
+        valuesToSave: {
+          ppudOffender: {
+            id: createOffenderResponse.offender.id,
+            croOtherNumber: recommendation.personOnProbation.croNumber,
+            dateOfBirth: recommendation.bookRecallToPpud?.dateOfBirth,
+            ethnicity: recommendation.bookRecallToPpud?.ethnicity,
+            familyName: recommendation.bookRecallToPpud?.lastName,
+            firstNames: recommendation.bookRecallToPpud?.firstNames,
+            gender: recommendation.bookRecallToPpud?.gender,
+            immigrationStatus: 'N/A',
+            nomsId: recommendation.personOnProbation?.nomsNumber,
+            prisonerCategory: 'N/A',
+            prisonNumber: recommendation.bookRecallToPpud?.prisonNumber,
```

Comment:
We don't need to have any references to immigrationStatus or prisonerCategory in CaR, PPUD Automation deals with that as it is always the same value.

Reply:
Same applies to youngOffender and status.